zcfan.c

```diff
@@ -225,6 +225,16 @@ static void maybe_ping_watchdog(void) {
         }                                                                      \
     } while (0)
 
+#define fscanf_str_for_key(f, pos, name, dest)                                 \
+    do {                                                                       \
+        char val[256];                                                         \
+        if (fscanf(f, name " %255s ", val) == 1) {                             \
```

Comment:
To avoid having to manually keep both 255 and 256 up to date, you can do something like:

```c
#define CONFIG_MAX_STRLEN 15
#define S_CONFIG_MAX_STRLEN STR(CONFIG_MAX_STRLEN)
char val[CONFIG_MAX_STRLEN + 1];
if (fscanf(f, name " %" S_CONFIG_MAX_STRLEN "s ", val) == 1)
```